pkg/operator/ceph/cluster/watcher.go

```diff
@@ -130,6 +170,122 @@ func (c *clientCluster) onK8sNode(ctx context.Context, object runtime.Object) bo
 	return false
 }
 
+func (c *clientCluster) createNetworkFence(ctx context.Context, node corev1.Node, clusterName string) error {
+	nodeIPtoBlock, pv, err := c.getWatcherIpPortToBlock(ctx, node, clusterName)
+	if err != nil {
+		return fmt.Errorf("failed to get node IP to block for node %q. %v", node.Name, err)
+	}
+	fmt.Println("NodeIP to block", nodeIPtoBlock)
+
+	networkFence := &addonsv1alpha1.NetworkFence{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      node.Name,
+			Namespace: c.namespace,
+		},
+		Spec: addonsv1alpha1.NetworkFenceSpec{
+			Driver:     "rook-ceph.rbd.csi.ceph.com",
+			FenceState: addonsv1alpha1.Fenced,
+			Secret: addonsv1alpha1.SecretSpec{
+				Name:      pv.Spec.CSI.ControllerPublishSecretRef.Name,
+				Namespace: pv.Spec.CSI.ControllerPublishSecretRef.Namespace,
+			},
+			Cidrs: []string{
+				nodeIPtoBlock,
+			},
+			Parameters: map[string]string{
+				"clusterID": pv.Spec.CSI.VolumeAttributes["clusterID"],
+			},
+		},
+	}
+
+	err = c.client.Create(ctx, networkFence)
+	if err != nil {
+		if errors.IsAlreadyExists(err) {
+			logger.Infof("network fence CR %s already exists", node.Name)
```

Comment:
> what if the old IP gets assigned to a new node later?

If the IP is not unblocklisted, then the new node would remain blocklisted. The user would need to unblocklist it. This seems like a corner enough case that we just document it, right?

Reply:
we can create the network fence cr with the combination of node Name and IP which we are blocking? It will help when we need to blocklist multiple IP on same node and will preserve the existing IP?

Reply:
As long as the csi addons supports blocklisting multiple IPs, and reconciling updates to add new IPs or remove unblocklisted IPs, I'd say let's stick with the node name as the CR name. Since only the IP is blocklisted at the ceph layer, I don't think we gain anything by adding the IP to the CR name.